src/realm/query_engine.hpp

```diff
@@ -1539,6 +1539,25 @@ class StringNodeBase : public ParentNode {
     {
         return m_leaf->get(s);
     }
+
+    size_t _find_first_local_interned_string(const StringData& data, size_t start, size_t end) const
+    {
+        REALM_ASSERT_DEBUG(m_leaf->is_interned());
```

Comment:
This will not hold.

Even though a string has been interned, we may have leafs with strings which are not interned (yet).

Consequently for every leaf a check has to be made and
A. if the leaf has been interned, we can compare integers
B. if the leaf has not been interned, we have to compare the strings, even though it is slower.